pkg/scheduler/preemption/preemption_test.go

```diff
@@ -43,17 +43,25 @@ import (
 	"sigs.k8s.io/kueue/pkg/constants"
 	controllerconstants "sigs.k8s.io/kueue/pkg/controller/constants"
 	"sigs.k8s.io/kueue/pkg/features"
-	"sigs.k8s.io/kueue/pkg/resources"
+	"sigs.k8s.io/kueue/pkg/hierarchy"
 	"sigs.k8s.io/kueue/pkg/scheduler/flavorassigner"
 	"sigs.k8s.io/kueue/pkg/util/slices"
 	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-var snapCmpOpts = cmp.Options{
-	// ignore zero values during comparison, as we consider
-	// zero FlavorResource usage to be same as no map entry.
-	cmpopts.IgnoreMapEntries(func(_ resources.FlavorResource, v int64) bool { return v == 0 }),
+func SnapCmpOpts() cmp.Options {
+	return cmp.Options{
+		cmp.AllowUnexported(
+			hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{},
+			cache.ClusterQueueSnapshot{},
+			cache.CohortSnapshot{},
```

Comment:
```suggestion
```

I think we don't need this. We already ignored it by `hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{},`.